components/Searchbar.js

```diff
@@ -0,0 +1,151 @@
+import Link from 'next/link';
+import { useRouter } from 'next/router';
+import { useRef, useState, useEffect } from 'react';
+
+export default function Searchbar({ books }) {
+	// Hooks for search bar
+	const searchRef = useRef(null);
+
+	// router
+	const router = useRouter();
+		
+	/**
+	 * @param {HTMLElement} child 
+	 * @param {HTMLElement} parent 
+	 */
+	/*
+	const isDeepChildOf = (child, parent) => {
+		console.log(child.parentNode, parent);
+		if (child == parent) {
+			return true;
+		} else if (child === document.body || child === document.documentElement || document.head || window) {
+			return false;
+		} else {
+			return isDeepChildOf(child.parentNode, parent);
+		}
+	}
+	*/
+
+	useEffect(() => {
+		window.addEventListener('keydown', (e) => {
+			if (e.key === '/' || (e.key === 'k' && e.ctrlKey)) {
+				if (searchRef.current && e.target !== searchRef.current) {
+					e.preventDefault();
+					searchRef.current.focus();
+				}
+			} else if (e.key === 'Escape') {
+				searchRef.current.value = '';
+				searchRef.current.blur();
+				setSearchItems('hidden');
+				setSelected(null);
+				setInputValue('');
+			}
+		});
+	}, []);
+
+	// list of books to be shown under search
+	const [searchItems, setSearchItems] = useState('hidden');
+	// selected book in list
+	const [selected, setSelected] = useState(null);
+	// current value of input
+	const [inputValue, setInputValue] = useState('');
+
+	return (
+		<div className="relative my-auto h-6 w-48">
+			<div
+				className='absolute ml-3 bg-white overflow-y-auto rounded-xl z-20 shadow-2xl'
+				onKeyDown={
+					(e) => {
+						if (searchItems instanceof Array && searchItems.length > 0) {
+							if (e.key === 'ArrowUp') {
+								e.preventDefault();
+								const idx = (typeof selected !== 'number') ? searchItems.length - 1 : (selected > 0) ? selected - 1 : searchItems.length - 1;
+								setSelected(idx);
+								searchRef.current.value = searchItems[idx].name;
+							} else if (e.key === 'ArrowDown') {
+								e.preventDefault();
+								const idx = (typeof selected !== 'number') ? 0 : (selected < searchItems.length - 1) ? selected + 1 : 0;
+								setSelected(idx);
+								searchRef.current.value = searchItems[idx].name;
+							} else if (e.key === 'Enter' && searchItems[selected]) {
+								router.push(`/${searchItems[selected].testament}/${searchItems[selected].name.toLowerCase()}`);
+							}
+						}
+					}
+				}
+			>
+				<input
+					onChange={
+						(e) => {
+							const v = e.target.value.toLowerCase();
```

Comment:
e is an okay variable because it is for the events. what is v?

Reply:
val? inputVal? inputValue?
These are so much better than v.

I understand e is ubiquitous for event, so that is the only reason e is okay.